libs/market-depth/src/lib/orderbook-data.ts

```diff
@@ -75,22 +76,32 @@ const updateRelativeData = (data: OrderbookRowData[]) => {
   });
 };
 
-export const createRow = (
+export const createPartialRow = (
   price: string,
   volume = 0,
   dataType?: VolumeType
-): OrderbookRowData => ({
+): PartialOrderbookRowData => ({
   price,
   ask: dataType === VolumeType.ask ? volume : 0,
   bid: dataType === VolumeType.bid ? volume : 0,
-  cumulativeVol: {
-    ask: dataType === VolumeType.ask ? volume : 0,
-    bid: dataType === VolumeType.bid ? volume : 0,
-  },
-  askByLevel: dataType === VolumeType.ask ? { [price]: volume } : {},
-  bidByLevel: dataType === VolumeType.bid ? { [price]: volume } : {},
 });
 
+export const extendRow = (row: PartialOrderbookRowData): OrderbookRowData =>
+  Object.assign(row, {
```

Comment:
nitpick (non-blocking): I normally prefer spread over assign

Reply:
Spread will create new object , and old one will need to be garbage collected. Object assign adds props to existing object.